### PR TITLE
Add ExperienceEvent application metrics

### PR DESCRIPTION
**Pull request by cdegroot-adobe**
Addition of standard mobile and application metrics.

### CODE REVIEW

File: schemas/data/metrics.schema.json [hunk 1]
```diff
@@ -31,12 +31,36 @@
         "https://ns.adobe.com/xdm/data/metrics/advertising/clicks": {
           "$ref": "https://ns.adobe.com/xdm/data/measure"
         },
```
Comment (lrosenthol) [inline schemas/data/metrics.schema.json]: can we get a description here?

File: schemas/data/metrics.schema.json [hunk 2]
```diff
@@ -31,12 +31,36 @@
...
+            "Activation of an application feature that is being measured."
+        },
+        "https://ns.adobe.com/xdm/data/metrics/application/installs": {
+          "$ref": "https://ns.adobe.com/xdm/data/measure",
+          "description":
+            "Install of an application on a device. Triggered on first launch after install."
+        },
         "https://ns.adobe.com/xdm/data/metrics/advertising/impressions": {
           "$ref": "https://ns.adobe.com/xdm/data/measure"
         },
```
Comment (lrosenthol) [inline schemas/data/metrics.schema.json]: Description?

File: schemas/data/metrics.schema.json [hunk 3]
```diff
@@ -31,12 +31,36 @@
...
         "https://ns.adobe.com/xdm/data/metrics/web/link-clicks": {
           "$ref": "https://ns.adobe.com/xdm/data/measure"
```
Comment (lrosenthol) [inline schemas/data/metrics.schema.json]: Description?

File: schemas/data/metrics.schema.json [hunk 4]
```diff
@@ -31,12 +31,36 @@
         "https://ns.adobe.com/xdm/data/metrics/advertising/clicks": {
           "$ref": "https://ns.adobe.com/xdm/data/measure"
         },
+        "https://ns.adobe.com/xdm/data/metrics/application/closes": {
```
Comment (lrosenthol) [inline schemas/data/metrics.schema.json]: Where are these application things coming from?  Is there some standard here - even within Adobe?  Does it come from a specific SDK?

File: schemas/data/metrics.schema.json [hunk 5]
```diff
@@ -31,12 +31,36 @@
...
           "$ref": "https://ns.adobe.com/xdm/data/measure"
         },
+        "https://ns.adobe.com/xdm/data/metrics/application/launches": {
+          "$ref": "https://ns.adobe.com/xdm/data/measure",
+          "description":
+            "A launch of an application. Triggered on every run, including crashes and installs. Also triggered on a resume from background when the lifecycle session timeout has been exceeded."
+        },
         "https://ns.adobe.com/xdm/data/metrics/direct-marketing/mirror-pages": {
           "$ref": "https://ns.adobe.com/xdm/data/measure"
         },
```
Comment (lrosenthol) [inline schemas/data/metrics.schema.json]: Description?

File: schemas/data/metrics.schema.json [hunk 6]
```diff
@@ -73,6 +97,11 @@
         "https://ns.adobe.com/xdm/data/metrics/direct-marketing/sends": {
           "$ref": "https://ns.adobe.com/xdm/data/measure"
         },
```
Comment (lrosenthol) [inline schemas/data/metrics.schema.json]: Description?

File: schemas/data/metrics.schema.json [hunk 7]
```diff
@@ -73,6 +97,11 @@
...
+        "https://ns.adobe.com/xdm/data/metrics/application/upgrades": {
+          "$ref": "https://ns.adobe.com/xdm/data/measure",
+          "description":
+            "Upgrade of an application that has previously been installed. Triggered on first launch after upgrade."
+        },
         "https://ns.adobe.com/xdm/data/metrics/direct-marketing/user-complaints": {
           "$ref": "https://ns.adobe.com/xdm/data/measure"
         }
```
Comment (lrosenthol) [inline schemas/data/metrics.schema.json]: Description?

File: schemas/data/metrics.schema.json (hunk 4 above)
Comment (cdegroot-adobe) [reply schemas/data/metrics.schema.json]: @lrosenthol  The primary library that is informing this set of additions in the Appmeasurement library for mobile (iOS/Android) with has been around for many years and is well adopted in our customer base. The metrics being added for applications are common across the industry for mobile applications.
Also there are a number of metrics tracked by creative cloud and doc cloud desktop applications when end users enroll in providing Adobe with this data. Most of the standard metrics line up well, the addition is featureUsage. 
One open item I am tracking is the concept of a installer download for desktop. More research is needed there to see if it aligns with a regular content download or is an application download is a different concept entirely.

**Comment by cdegroot-adobe** [comment]
@lrosenthol - In regard to the missing descriptions from the metrics not in this PR, I have created an issue to track and will fix that as a dedicated PR.

https://github.com/adobe/xdm/issues/226

It seems that keeping PRs small and specific keeps things moving faster.

File: schemas/data/metrics.schema.json (hunk 4 above)
Comment (lrosenthol) [reply schemas/data/metrics.schema.json]: Thanks for the info!

**Comment by lrosenthol** [comment]
in general @cdegroot-adobe, I would agree about smaller issues, but in this case, I don't want to merge this without the descriptions - so easier to just do them here...

**Comment by cdegroot-adobe** [comment]
@kstreeter @lrosenthol 
Hello gents, I have updated all the descriptions and have cleaned up a couple other little items in the previous text, I think it is good now. Please review and merge or let me know any issues you see and I will fix tomorrow.

File: schemas/data/metrics.schema.json [hunk 8]
```diff
@@ -20,61 +20,123 @@
...
+        "https://ns.adobe.com/xdm/data/metrics/application/upgrades": {
+          "$ref": "https://ns.adobe.com/xdm/data/measure",
+          "description":
+            "Upgrade of an application that has previously been installed. Triggered on first launch after upgrade."
         },
         "https://ns.adobe.com/xdm/data/metrics/direct-marketing/user-complaints": {
-          "$ref": "https://ns.adobe.com/xdm/data/measure"
+          "$ref": "https://ns.adobe.com/xdm/data/measure",
+          "description":
+            "User complaints has been received. This generally occurs when a recipient of a message reports it as spam."
```
Comment (kstreeter) [inline schemas/data/metrics.schema.json]: typo has->have

**Comment by kstreeter** [comment]
@cdegroot-adobe I had one comment on a type, otherwise I am good with this.

**Comment by trieloff** [comment]
@lrosenthol have your concerns been addressed?

**Comment by cdegroot-adobe** [comment]
@kstreeter can we merge this in so I can then do an update to the metric schema.

**Pull request merged**